plugins/velocity/src/main/java/org/apache/struts2/views/velocity/VelocityManager.java

```diff
@@ -161,32 +164,24 @@ public Context createContext(ValueStack stack, HttpServletRequest req, HttpServl
      * perform any initialization of the contexts.  All that must be done in the
      * context itself.
      *
-     * @param servletRequest the servlet request object
+     * @param servletRequest  the servlet request object
      * @param servletResponse the servlet response object
-     * @param extraContext map with extra context
+     * @param extraContext    map with extra context
      * @return an VelocityContext[] of contexts to chain
      */
-    protected VelocityContext[] prepareChainedContexts(HttpServletRequest servletRequest, HttpServletResponse servletResponse, Map extraContext) {
-        if (this.chainedContextNames == null) {
-            return null;
-        }
-        List contextList = new ArrayList();
-        for (int i = 0; i < chainedContextNames.length; i++) {
-            String className = chainedContextNames[i];
-            try {
-                VelocityContext velocityContext = (VelocityContext) objectFactory.buildBean(className, null);
-                contextList.add(velocityContext);
-            } catch (Exception e) {
-                LOG.warn("Warning. {} caught while attempting to instantiate a chained VelocityContext, {} -- skipping", e.getClass().getName(), className);
+    protected List<VelocityContext> prepareChainedContexts(HttpServletRequest servletRequest, HttpServletResponse servletResponse, Map<String, Object> extraContext) {
```

Comment:
Similar to the constructor involving `VelocityContext[]`, since this method is `protected`, changing the return type might introduce a backwards-compatibility issue for someone (but less likely than the constructor).

Looks like there's no way to avoid this unless the new method uses a different parameter signature than the old one does.  One (ugly) way this might work would be to do something like  this for a backwards-compatible version:
```
// Reorder parameters in new method (different signature from old)
protected List<VelocityContext> prepareChainedContexts(Map<String, Object> extraContext, HttpServletRequest servletRequest, HttpServletResponse servletResponse) {
    ... // The rest of the new method
}

protected VelocityContext[] prepareChainedContexts(HttpServletRequest servletRequest, HttpServletResponse servletResponse, Map<String, Object> extraContext) {
    return prepareChainedContexts(extraContext, servletRequest, servletResponse).toArray();
}
// And change the other internal calls to use the new parameter order, just like the backwards-compatible call.
```

It is _probably not worth the hassle_, but I figured it was worth mentioning.  :smile:

Reply:
Yeah... it's hard to keep backward-compatibility all the time, that's why those changes were introduced in a major Struts version 2.6.

Reply:
OK.  If it really creates an issue for someone in 2.6 they can always propose a solution.  :smile:
Interesting to note the IDEA IDE formatting behaviour as well.